packages/react-dom/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -749,7 +749,7 @@ function pushStartOption(
     }
   }
 
-  if (selectedValue !== null) {
+  if (selectedValue != null) {
```

Comment:
The type of `selectedValue` is a lie. We should probably toString it and resolve the undefined earlier.